pkg/dependency/parser/java/pom/parse.go

```diff
@@ -312,7 +318,7 @@ func (p *parser) analyze(pom *pom, opts analysisOptions) (analysisResult, error)
 	}
 
 	// Update remoteRepositories
-	p.remoteRepositories = utils.UniqueStrings(append(p.remoteRepositories, pom.repositories()...))
+	p.remoteRepositories = utils.UniqueStrings(append(p.remoteRepositories, pom.repositories(p.servers)...))
```

Comment:
I found that we are already adding repositories for **every** pom file (child, parent pom) we analyze.
I added a server check.

Reply:
@DmitriyLewen I have changed the order of the repo to make the first try with the repo in the pom.xml to reduce the number of tries